client_test.go

```diff
@@ -511,74 +509,73 @@ func Test_Client(t *testing.T) {
 		config.Queues["alternate"] = QueueConfig{MaxWorkers: 10}
 		client := newTestClient(t, bundle.dbPool, config)
 
-		jobStartedChan := make(chan int64)
-
-		type JobArgs struct {
-			JobArgsReflectKind[JobArgs]
-		}
-
-		AddWorker(client.config.Workers, WorkFunc(func(ctx context.Context, job *Job[JobArgs]) error {
-			jobStartedChan <- job.ID
-			return nil
-		}))
-
+		subscribeChan := subscribe(t, client)
 		startClient(ctx, t, client)
 
-		client.producersByQueueName[QueueDefault].testSignals.Init()
-		client.producersByQueueName["alternate"].testSignals.Init()
-
-		insertRes1, err := client.Insert(ctx, &JobArgs{}, nil)
+		insertRes1, err := client.Insert(ctx, &noOpArgs{}, nil)
 		require.NoError(t, err)
 
-		startedJobID := riverinternaltest.WaitOrTimeout(t, jobStartedChan)
-		require.Equal(t, insertRes1.Job.ID, startedJobID)
+		event := riverinternaltest.WaitOrTimeout(t, subscribeChan)
+		require.Equal(t, EventKindJobCompleted, event.Kind)
+		require.Equal(t, insertRes1.Job.ID, event.Job.ID)
```

Comment:
Just for the sake of keeping these tests shorter, what do you think of only subscribing to queue-based events here, then we can skip the checks on job events?

Reply:
With the rebased version here, I don't think subscribing to only queue events would give us coverage around when the jobs are handled (i.e. not while paused).

Reply:
Cool. Yeah I guess it reduces some of the other boiler plate, so seems good.